test/integration/gardenlet/tokenrequestor/workloadidentity/workloadidentity_suite_test.go

```diff
@@ -0,0 +1,174 @@
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+
+	securityv1alpha1 "github.com/gardener/gardener/pkg/apis/security/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/gardenlet/controller/tokenrequestor/workloadidentity"
+	"github.com/gardener/gardener/pkg/logger"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	gardenerenvtest "github.com/gardener/gardener/test/envtest"
+)
+
+func TestTokenRequestorWorkloadIdentity(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test Integration Gardenlet TokenRequestor WorkloadIdentity Suite")
+}
+
+const testID = "tokenrequestor-workloadidentity-controller-test"
+
+var (
+	ctx = context.Background()
+	log logr.Logger
+
+	restConfig *rest.Config
+	testEnv    *gardenerenvtest.GardenerTestEnvironment
+	testClient client.Client
+
+	testNamespace *corev1.Namespace
+	testRunID     string
+
+	fakeClock       *testclock.FakeClock
+	verificationKey crypto.PublicKey
+)
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(logger.MustNewZapLogger(logger.DebugLevel, logger.FormatJSON, zap.WriteTo(GinkgoWriter)))
+	log = logf.Log.WithName(testID)
+
+	dirPath := filepath.Join(".", "testdata")
+	if err := os.Mkdir(dirPath, 0700); err != nil && !os.IsExist(err) {
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	keyFile := filepath.Join(dirPath, "key.pem")
+	if _, err := os.Stat(keyFile); os.IsNotExist(err) {
+		key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+		Expect(err).ToNot(HaveOccurred())
+		verificationKey = key.Public()
+		bytes, err := x509.MarshalECPrivateKey(key)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(os.WriteFile(keyFile, pem.EncodeToMemory(&pem.Block{
+			Type:  "EC PRIVATE KEY",
+			Bytes: bytes,
+		}), 0600)).To(Succeed())
+	} else {
+		keyData, err := os.ReadFile(keyFile)
+		Expect(err).ToNot(HaveOccurred())
+		keyBlock, _ := pem.Decode(keyData)
+		key, err := x509.ParseECPrivateKey(keyBlock.Bytes)
+		Expect(err).ToNot(HaveOccurred())
+		verificationKey = key.Public()
+	}
+
+	By("Start test environment")
+	testEnv = &gardenerenvtest.GardenerTestEnvironment{
+		Environment: &envtest.Environment{
+			CRDInstallOptions: envtest.CRDInstallOptions{
+				Paths: []string{},
+			},
+			ErrorIfCRDPathMissing: true,
+		},
+		GardenerAPIServer: &gardenerenvtest.GardenerAPIServer{
+			Args: []string{
+				"--disable-admission-plugins=DeletionConfirmation,Bastion,ResourceReferenceManager,ExtensionValidator,ShootDNS,ShootQuotaValidator,ShootTolerationRestriction,ShootValidator,SeedValidator",
+				"--workload-identity-token-issuer=https://local.gardener.cloud",
+				"--workload-identity-signing-key-file=" + keyFile,
+			},
+		},
+	}
+
+	var err error
+	restConfig, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(restConfig).NotTo(BeNil())
+
+	DeferCleanup(func() {
```

Comment:
It would be nice to delete the `key.pem` file here, otherwise once written on the file system it will never be re-generated (should affect only developers) and this could cause issues in the future if the current key type stop being supported.